inbm/dispatcher-agent/dispatcher/update_logger.py

```diff
@@ -0,0 +1,76 @@
+"""
+    Class for creating UpdateLogger objects to record OTA update status
+
+    Copyright (C) 2017-2023 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+
+import datetime
+import logging
+from typing import Optional
+
+from inbm_lib.constants import LOG_FILE, OTA_PENDING
+
+logger = logging.getLogger(__name__)
+
+
+class UpdateLogger:
+    """UpdateLogger class stores the OTA update information and generates
+       log file.
+    @param type: Command to be executed
+    @param data: meta-data of the OTA
+    """
+
+    def __init__(self, ota_type: Optional[str], data: Optional[str]) -> None:
+        self._status = ""
+        self._ota_type = ota_type
+        self._time = datetime.datetime.now()
+        self._meta_data = data
+        self._error: Optional[str] = None
+
+    def set_time(self) -> None:
+        """Set the OTA starting time."""
+        self._time = datetime.datetime.now()
+
+    def set_status_and_error(self, status: str, err: Optional[str]) -> None:
+        """Set status and error message."""
```

Comment:
Coding standards issue.  Missing the description, etc for all the parameters.  We'll need to address this in a commit after the drop.

Reply:
Or if these aren't called outside of this class, they need to be made private instead of public.